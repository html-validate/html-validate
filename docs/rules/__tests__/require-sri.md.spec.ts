import HtmlValidate from "../../../src/htmlvalidate";

const markup: { [key: string]: string } = {};
markup["incorrect"] = `<script href="//cdn.example.net/jquery.min.js"></script>`;
markup["correct"] = `<script href="//cdn.example.net/jquery.min.js" integrity="sha384-..."></script>`;
markup["crossorigin"] = `<!--- local resource -->
<link href="local.css">

<!-- resource loaded over CDN -->
<link href="//cdn.example.net/remote.css">`;

describe("docs/rules/require-sri.md", () => {
	it("inline validation: incorrect", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"require-sri":"error"}});
		const report = htmlvalidate.validateString(markup["incorrect"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: correct", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"require-sri":"error"}});
		const report = htmlvalidate.validateString(markup["correct"]);
		expect(report.results).toMatchSnapshot();
	});
	it("inline validation: crossorigin", () => {
		expect.assertions(1);
		const htmlvalidate = new HtmlValidate({"rules":{"require-sri":["error",{"target":"crossdomain"}]}});
		const report = htmlvalidate.validateString(markup["crossorigin"]);
		expect(report.results).toMatchSnapshot();
	});
});
