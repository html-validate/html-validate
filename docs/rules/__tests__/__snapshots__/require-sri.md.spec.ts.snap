// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`docs/rules/require-sri.md inline validation: correct 1`] = `Array []`;

exports[`docs/rules/require-sri.md inline validation: crossorigin 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "inline",
    "messages": Array [
      Object {
        "column": 2,
        "context": undefined,
        "line": 5,
        "message": "SRI \\"integrity\\" attribute is required on <link> element",
        "offset": 85,
        "ruleId": "require-sri",
        "severity": 2,
        "size": 4,
      },
    ],
    "source": "<!--- local resource -->
<link href=\\"local.css\\">

<!-- resource loaded over CDN -->
<link href=\\"//cdn.example.net/remote.css\\">",
    "warningCount": 0,
  },
]
`;

exports[`docs/rules/require-sri.md inline validation: incorrect 1`] = `
Array [
  Object {
    "errorCount": 1,
    "filePath": "inline",
    "messages": Array [
      Object {
        "column": 2,
        "context": undefined,
        "line": 1,
        "message": "SRI \\"integrity\\" attribute is required on <script> element",
        "offset": 1,
        "ruleId": "require-sri",
        "severity": 2,
        "size": 6,
      },
    ],
    "source": "<script href=\\"//cdn.example.net/jquery.min.js\\"></script>",
    "warningCount": 0,
  },
]
`;
